tests/unittests/config/test_cc_set_passwords.py

```diff
@@ -376,6 +527,96 @@ class TestSetPasswordsSchema:
+                                "type": "text",
+                                "password": "correct-horse-battery-staple",
+                            },
+                            {
+                                "name": "what-if-2",
+                                "type": "hash",
+                                "password": "no-magic-parsing-done-here",
+                            },
+                            {
+                                "name": "what-if-3",
+                                "password": "type-is-optional-default-"
+                                "value-is-hash",
+                            },
+                            {
+                                "name": "what-if-4",
+                                "type": "RANDOM",
+                            },
+                        ]
+                    }
+                },
+                None,
+            ),
+            (
+                {
+                    "chpasswd": {
+                        "users": [
+                            {
+                                "name": "what-if-1",
+                                "type": "plaintext",
+                                "password": "type-has-two-legal-values: "
+                                "{'hash', 'text'}",
+                            }
+                        ]
+                    }
+                },
+                "is not valid under any of the given schemas",
+            ),
+            (
+                {
+                    "chpasswd": {
+                        "users": [
+                            {
+                                "name": "what-if-1",
+                                "type": "RANDOM",
+                                "password": "but you want random?",
+                            }
+                        ]
+                    }
+                },
+                "is not valid under any of the given schemas",
+            ),
+            (
+                {"chpasswd": {"users": [{"password": "."}]}},
+                "is not valid under any of the given schemas",
+            ),
+            # when type != RANDOM, password is a required key
+            (
+                {
+                    "chpasswd": {
+                        "users": [{"name": "what-if-1", "type": "hash"}]
+                    }
+                },
+                "is not valid under any of the given schemas",
+            ),
+            # dat is an additional property
+            (
+                {
+                    "chpasswd": {
+                        "users": [
+                            {
+                                "name": "sonata",
+                                "password": "dit",
+                                "dat": "dot",
+                            }
+                        ]
+                    }
+                },
+                "is not valid under any of the given schemas",
+            ),
```

Comment:
One thing that is a nice-to-have from unittest run perspective is aptly named test ids for these cases which allows us to avoid the comment, and we get to see specfic labelled test ids if they fall.
you get to see womthing like:
```test_cc_set_passwords.py::TestSetPasswordsSchema::test_schema_validation[dat_is_an_additional_property]```
instead of 
```test_cc_set_passwords.py::TestSetPasswordsSchema::test_schema_validation[config8-is not valid under any of the given schemas```

You can to something like this:
```suggestion
            pytest.param(
                {
                    "chpasswd": {
                        "users": [
                            {
                                "name": "sonata",
                                "password": "dit",
                                "dat": "dot",
                            }
                        ]
                    }
                },
                "is not valid under any of the given schemas",
                id="dat_is_an_additional_property"
            )
```